### ISSUE TITLE
[dev.icinga.com #3972] Role creation fails with 'invalid parameter number bound'

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga-web#1103]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/3972

**Created by mfrosch on 2013-04-16 19:53:11 +00:00**

Assignee: _mfrosch_
Status: _Resolved_  (closed on _2013-04-16 22:45:49 +00:00_)
Target Version: _1.9_
Last Update: _2013-04-16 22:45:49 +00:00 (in Redmine)_

    Icinga Version: 1.8.4
    Icinga Web Version: 1.9.0-dev
    IDO Version: 1.8.4
    OS Version: Debian/wheezy
    DB Type: UNKNOWN
    DB Version: all
    Browser Version: all

---

The changes made with [\#3959](https://dev.icinga.com/issues/3959) causes this error.

Basically the Api tries to UPDATE only, but can't handle INSERT, which was handled by save() before.

**Changesets**

_2013-04-16 19:53:21 +00:00_ by _mfrosch_ 161887ba02898dfafd3808e239f25cc9543f74da

    Fix role creation
    
    Role creation failed with 'invalid parameter number bound'
    
    There was no insert since the changes of (refs #3959)
    
    re-implemented when role_id === null
    
    fixes #3972

---

**Relations**:

* relates [#3959](https://dev.icinga.com/issues/3959)
